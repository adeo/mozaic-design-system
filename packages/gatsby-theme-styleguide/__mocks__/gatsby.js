const React = require(`react`)
const gatsby = jest.requireActual(`gatsby`)

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
}
