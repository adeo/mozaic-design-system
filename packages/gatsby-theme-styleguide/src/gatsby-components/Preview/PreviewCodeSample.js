import React from 'react'
import styled, { css } from 'styled-components'

const CodeButton = styled.button`
  ${isActive =>
    isActive ? `background-color: lightblue;` : `background-color: #999;`};
`
const Preformated = styled.pre`
  max-height: 600px;
  max-width: 100%;
  overflow: auto;

  ${({ fullScreen }) =>
    fullScreen &&
    css`
      width: 100%;
      max-height: none;
      overflow: none;

      code {
        overflow: none;
      }
    `};
`

const Body = styled.div`
  flex: 1;
  overflow: auto;
`

const PreviewCodeSample = ({
  preview,
  copyCompatible,
  showCode,
  getLanguage,
  currentCodeSample,
  copied,
  copyCodeToClipBoard,
  fullScreen,
}) => {
  return (
    <>
      <div>
        {Object.keys(preview).map(codeSampleName => (
          <CodeButton
            isActive={codeSampleName === currentCodeSample}
            key={`${preview}${codeSampleName}`}
            onClick={() => showCode(codeSampleName)}
          >
            {codeSampleName}
          </CodeButton>
        ))}
        {copyCompatible && (
          <button style={{ float: 'right' }} onClick={copyCodeToClipBoard}>
            {copied ? 'copied' : 'copy code'}
          </button>
        )}
      </div>

      <Body>
        {currentCodeSample && (
          <Preformated fullScreen={fullScreen}>
            <code className={`language-${getLanguage(currentCodeSample)}`}>
              {preview[currentCodeSample]}
            </code>
          </Preformated>
        )}
      </Body>
    </>
  )
}
export default PreviewCodeSample
