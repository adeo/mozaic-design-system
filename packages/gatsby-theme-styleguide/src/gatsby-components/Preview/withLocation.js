import React from 'react'
import Preview from './Preview'

export default location => {
  return props => <Preview {...props} location={location} />
}
