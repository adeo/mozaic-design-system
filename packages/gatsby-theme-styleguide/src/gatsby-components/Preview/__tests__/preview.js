import React from 'react'
const { mount, shallow } = require('enzyme')
import { StaticQuery, query } from 'gatsby'

import Preview from '../Preview'

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      allPreview: {
        edges: [
          {
            node: {
              id: '3d888a2b-320e-5698-8933-13c3df3b1621',
              path: 'src/docs/Components/Buttons/previews/button-icon-position',
              codes: {
                js: '',
                html:
                  '<div class="example">\n  <a href="#" class="mc-button">\n    <svg class="mc-button__icon  mc-button__icon--m">\n      <use xlink:href="#icon" />\n    </svg>\n    <span class="mc-button__label">\n      Icon on the left\n    </span>\n  </a>\n  <a href="#" class="mc-button">\n    <span class="mc-button__label">\n      Icon on the right\n    </span>\n    <svg class="mc-button__icon  mc-button__icon--m">\n      <use xlink:href="#icon" />\n    </svg>\n  </a>\n</div>\n\n<svg xmlns="http://www.w3.org/2000/svg" style="display: none">\n  <symbol id="icon" viewBox="0 0 24 24">\n    <path\n      d="M17.13 16.57H8a1 1 0 0 1-1-.74L4.39 6H3a1 1 0 0 1 0-2h2.15a1 1 0 0 1 1 .74l.53 2H20A1 1 0 0 1 20.94 8l-2.87 7.87a1 1 0 0 1-.94.7zm-8.34-2h7.64l2.14-5.87H7.2z"\n    />\n    <circle cx="9.46" cy="19.85" r="2.15" />\n    <circle cx="15.69" cy="19.85" r="2.15" />\n  </symbol>\n</svg>\n',
                scss:
                  "@import 'settings-tools/_all-settings';\n\n@include import-font-families();\n\n@import 'components/_c.button';\n\n.example {\n  text-align: center;\n  margin: $mu100 auto;\n}\n",
                css:
                  '/* create columns */\n/* create columns */\n/* create custom named columns with custom content */\n@font-face {\n  font-family: \'LeroyMerlin\';\n  src: url("/fonts/LeroyMerlinSans-Web-Light.woff2") format("woff2"), url("/fonts/LeroyMerlinSans-Web-Light.woff") format("woff");\n  font-weight: 300;\n  font-style: normal;\n}\n\n@font-face {\n  font-family: \'LeroyMerlin\';\n  src: url("/fonts/LeroyMerlinSans-Web-Regular.woff2") format("woff2"), url("/fonts/LeroyMerlinSans-Web-Regular.woff") format("woff");\n  font-weight: 400;\n  font-style: normal;\n}\n\n@font-face {\n  font-family: \'LeroyMerlin\';\n  src: url("/fonts/LeroyMerlinSans-Web-SemiBold.woff2") format("woff2"), url("/fonts/LeroyMerlinSans-Web-SemiBold.woff") format("woff");\n  font-weight: 600;\n  font-style: normal;\n}\n\n@font-face {\n  font-family: \'LeroyMerlin\';\n  src: url("/fonts/LeroyMerlinSans-Web-LightItalic.woff2") format("woff2"), url("/fonts/LeroyMerlinSans-Web-LightItalic.woff") format("woff");\n  font-weight: 300;\n  font-style: italic;\n}\n\n@font-face {\n  font-family: \'LeroyMerlin\';\n  src: url("/fonts/LeroyMerlinSans-Web-Italic.woff2") format("woff2"), url("/fonts/LeroyMerlinSans-Web-Italic.woff") format("woff");\n  font-weight: 400;\n  font-style: italic;\n}\n\n@font-face {\n  font-family: \'LeroyMerlin\';\n  src: url("/fonts/LeroyMerlinSans-Web-SemiBoldItalic.woff2") format("woff2"), url("/fonts/LeroyMerlinSans-Web-SemiBoldItalic.woff") format("woff");\n  font-weight: 600;\n  font-style: italic;\n}\n.mc-button {\n  margin: 0;\n  -webkit-box-shadow: none;\n          box-shadow: none;\n  text-decoration: none;\n  outline: none;\n  background: none;\n  border: none;\n  padding: 0;\n  cursor: pointer;\n  color: #ffffff;\n  fill: currentColor;\n  background-color: #78be20;\n  font-family: \'LeroyMerlin\', sans-serif;\n  font-weight: 600;\n  font-size: 1rem;\n  line-height: 1;\n  padding: 0.25rem 2rem;\n  min-height: 3rem;\n  min-width: 3rem;\n  /* ie11 hack to center vertically */\n  /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n  height: 0;\n  cursor: pointer;\n  border-radius: 4px;\n  text-align: center;\n  border: 2px solid transparent;\n  -webkit-transition: all ease 200ms;\n  -o-transition: all ease 200ms;\n  transition: all ease 200ms;\n  display: -webkit-inline-box;\n  display: -ms-inline-flexbox;\n  display: inline-flex;\n  -webkit-box-align: center;\n      -ms-flex-align: center;\n          align-items: center;\n  -webkit-box-pack: center;\n      -ms-flex-pack: center;\n          justify-content: center;\n  vertical-align: middle;\n  -webkit-box-sizing: border-box;\n          box-sizing: border-box;\n}\n\n.mc-button:hover, .mc-button.is-hover {\n  background-color: #41a017;\n  color: #ffffff;\n}\n\n.mc-button:hover, .mc-button.is-hover {\n  background-color: #41a017;\n  color: #ffffff;\n}\n\n.mc-button:focus, .mc-button.is-focus {\n  border-color: #158110;\n}\n\n.mc-button:active, .mc-button.is-active {\n  background-color: #158110;\n}\n\n.mc-button:disabled, .mc-button.is-disabled {\n  border-color: transparent;\n  background-color: #eeeef0;\n  color: #887f87;\n  cursor: default;\n}\n\n.mc-button:focus, .mc-button.is-focus {\n  -webkit-box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.2);\n          box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.2);\n}\n\n.mc-button--s {\n  font-size: 0.875rem;\n  line-height: 1;\n  padding: 0.25rem 1rem;\n  min-height: 2rem;\n  min-width: 2rem;\n  /* ie11 hack to center vertically */\n  /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n  height: 0;\n}\n\n.mc-button--m {\n  font-size: 1rem;\n  line-height: 1;\n  padding: 0.25rem 2rem;\n  min-height: 3rem;\n  min-width: 3rem;\n  /* ie11 hack to center vertically */\n  /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n  height: 0;\n}\n\n.mc-button--l {\n  font-size: 1.125rem;\n  line-height: 1;\n  padding: 0.25rem 2rem;\n  min-height: 3.5rem;\n  min-width: 3.5rem;\n  /* ie11 hack to center vertically */\n  /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n  height: 0;\n}\n\n.mc-button--fit {\n  display: -webkit-inline-box;\n  display: -ms-inline-flexbox;\n  display: inline-flex;\n}\n\n.mc-button--full {\n  display: -webkit-box;\n  display: -ms-flexbox;\n  display: flex;\n}\n\n.mc-button--square {\n  padding: 0;\n}\n\n.mc-button__icon:last-child {\n  margin-left: 0.5rem;\n  margin-right: -0.25rem;\n}\n\n.mc-button__icon:first-child {\n  margin-right: 0.5rem;\n  margin-left: -0.25rem;\n}\n\n.mc-button__icon:only-child {\n  margin: 0;\n}\n\n.mc-button__icon--s {\n  width: 1rem;\n  height: 1rem;\n}\n\n.mc-button__icon--m {\n  width: 1.5rem;\n  height: 1.5rem;\n}\n\n.mc-button__icon--l {\n  width: 2rem;\n  height: 2rem;\n}\n\n.mc-button__icon--xl {\n  width: 2.5rem;\n  height: 2.5rem;\n}\n\n.mc-button__label {\n  -webkit-transform: translateY(-0.125em);\n      -ms-transform: translateY(-0.125em);\n          transform: translateY(-0.125em);\n}\n\n.mc-button--solid-primary-02 {\n  color: #ffffff;\n  fill: currentColor;\n  background-color: #007574;\n}\n\n.mc-button--solid-primary-02:hover, .mc-button--solid-primary-02.is-hover {\n  background-color: #063a44;\n  color: #ffffff;\n}\n\n.mc-button--solid-primary-02:hover, .mc-button--solid-primary-02.is-hover {\n  background-color: #063a44;\n  color: #ffffff;\n}\n\n.mc-button--solid-primary-02:focus, .mc-button--solid-primary-02.is-focus {\n  border-color: #062b35;\n}\n\n.mc-button--solid-primary-02:active, .mc-button--solid-primary-02.is-active {\n  background-color: #062b35;\n}\n\n.mc-button--solid-primary-02:disabled, .mc-button--solid-primary-02.is-disabled {\n  border-color: transparent;\n  background-color: #eeeef0;\n  color: #887f87;\n  cursor: default;\n}\n\n.mc-button--solid-neutral {\n  color: #ffffff;\n  fill: currentColor;\n  background-color: #887f87;\n}\n\n.mc-button--solid-neutral:hover, .mc-button--solid-neutral.is-hover {\n  background-color: #554f52;\n  color: #ffffff;\n}\n\n.mc-button--solid-neutral:hover, .mc-button--solid-neutral.is-hover {\n  background-color: #554f52;\n  color: #ffffff;\n}\n\n.mc-button--solid-neutral:focus, .mc-button--solid-neutral.is-focus {\n  border-color: #222020;\n}\n\n.mc-button--solid-neutral:active, .mc-button--solid-neutral.is-active {\n  background-color: #3c3738;\n}\n\n.mc-button--solid-neutral:disabled, .mc-button--solid-neutral.is-disabled {\n  border-color: transparent;\n  background-color: #eeeef0;\n  color: #887f87;\n  cursor: default;\n}\n\n.mc-button--solid-danger {\n  color: #ffffff;\n  fill: currentColor;\n  background-color: #df382b;\n}\n\n.mc-button--solid-danger:hover, .mc-button--solid-danger.is-hover {\n  background-color: #b42a27;\n  color: #ffffff;\n}\n\n.mc-button--solid-danger:hover, .mc-button--solid-danger.is-hover {\n  background-color: #b42a27;\n  color: #ffffff;\n}\n\n.mc-button--solid-danger:focus, .mc-button--solid-danger.is-focus {\n  border-color: #8b2226;\n}\n\n.mc-button--solid-danger:active, .mc-button--solid-danger.is-active {\n  background-color: #8b2226;\n}\n\n.mc-button--solid-danger:disabled, .mc-button--solid-danger.is-disabled {\n  border-color: transparent;\n  background-color: #eeeef0;\n  color: #887f87;\n  cursor: default;\n}\n\n.mc-button--bordered {\n  color: #78be20;\n  fill: currentColor;\n  border: 2px solid #78be20;\n  background-color: #ffffff;\n}\n\n.mc-button--bordered:hover, .mc-button--bordered.is-hover {\n  background-color: #eaf3e2;\n  color: #78be20;\n}\n\n.mc-button--bordered:focus, .mc-button--bordered.is-focus {\n  border-color: #158110;\n}\n\n.mc-button--bordered:active, .mc-button--bordered.is-active {\n  background-color: #cbe3b5;\n}\n\n.mc-button--bordered:disabled, .mc-button--bordered.is-disabled {\n  border-color: transparent;\n  background-color: #eeeef0;\n  color: #887f87;\n  cursor: default;\n}\n\n.mc-button--bordered-primary-02 {\n  color: #007574;\n  fill: currentColor;\n  border: 2px solid #007574;\n  background-color: #ffffff;\n}\n\n.mc-button--bordered-primary-02:hover, .mc-button--bordered-primary-02.is-hover {\n  background-color: #dbedea;\n  color: #007574;\n}\n\n.mc-button--bordered-primary-02:focus, .mc-button--bordered-primary-02.is-focus {\n  border-color: #062b35;\n}\n\n.mc-button--bordered-primary-02:active, .mc-button--bordered-primary-02.is-active {\n  background-color: #a5d1cb;\n}\n\n.mc-button--bordered-primary-02:disabled, .mc-button--bordered-primary-02.is-disabled {\n  border-color: transparent;\n  background-color: #eeeef0;\n  color: #887f87;\n  cursor: default;\n}\n\n.mc-button--bordered-neutral {\n  color: #554f52;\n  fill: currentColor;\n  border: 2px solid #554f52;\n  background-color: #ffffff;\n}\n\n.mc-button--bordered-neutral:hover, .mc-button--bordered-neutral.is-hover {\n  background-color: #eeeef0;\n  color: #554f52;\n}\n\n.mc-button--bordered-neutral:focus, .mc-button--bordered-neutral.is-focus {\n  border-color: #222020;\n}\n\n.mc-button--bordered-neutral:active, .mc-button--bordered-neutral.is-active {\n  background-color: #d3d2d6;\n}\n\n.mc-button--bordered-neutral:disabled, .mc-button--bordered-neutral.is-disabled {\n  border-color: transparent;\n  background-color: #eeeef0;\n  color: #887f87;\n  cursor: default;\n}\n\n.mc-button--bordered-danger {\n  color: #df382b;\n  fill: currentColor;\n  border: 2px solid #df382b;\n  background-color: #ffffff;\n}\n\n.mc-button--bordered-danger:hover, .mc-button--bordered-danger.is-hover {\n  background-color: #feedee;\n  color: #df382b;\n}\n\n.mc-button--bordered-danger:focus, .mc-button--bordered-danger.is-focus {\n  border-color: #8b2226;\n}\n\n.mc-button--bordered-danger:active, .mc-button--bordered-danger.is-active {\n  background-color: #fab9bc;\n}\n\n.mc-button--bordered-danger:disabled, .mc-button--bordered-danger.is-disabled {\n  border-color: transparent;\n  background-color: #eeeef0;\n  color: #887f87;\n  cursor: default;\n}\n\n@media screen and (min-width: 680px) {\n  .mc-button--s\\@from-m {\n    font-size: 0.875rem;\n    line-height: 1;\n    padding: 0.25rem 1rem;\n    min-height: 2rem;\n    min-width: 2rem;\n    /* ie11 hack to center vertically */\n    /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n    height: 0;\n  }\n  .mc-button--m\\@from-m {\n    font-size: 1rem;\n    line-height: 1;\n    padding: 0.25rem 2rem;\n    min-height: 3rem;\n    min-width: 3rem;\n    /* ie11 hack to center vertically */\n    /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n    height: 0;\n  }\n  .mc-button--l\\@from-m {\n    font-size: 1.125rem;\n    line-height: 1;\n    padding: 0.25rem 2rem;\n    min-height: 3.5rem;\n    min-width: 3.5rem;\n    /* ie11 hack to center vertically */\n    /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n    height: 0;\n  }\n  .mc-button--fit\\@from-m {\n    display: -webkit-inline-box;\n    display: -ms-inline-flexbox;\n    display: inline-flex;\n  }\n  .mc-button--full\\@from-m {\n    display: -webkit-box;\n    display: -ms-flexbox;\n    display: flex;\n  }\n}\n\n@media screen and (min-width: 1024px) {\n  .mc-button--s\\@from-l {\n    font-size: 0.875rem;\n    line-height: 1;\n    padding: 0.25rem 1rem;\n    min-height: 2rem;\n    min-width: 2rem;\n    /* ie11 hack to center vertically */\n    /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n    height: 0;\n  }\n  .mc-button--m\\@from-l {\n    font-size: 1rem;\n    line-height: 1;\n    padding: 0.25rem 2rem;\n    min-height: 3rem;\n    min-width: 3rem;\n    /* ie11 hack to center vertically */\n    /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n    height: 0;\n  }\n  .mc-button--l\\@from-l {\n    font-size: 1.125rem;\n    line-height: 1;\n    padding: 0.25rem 2rem;\n    min-height: 3.5rem;\n    min-width: 3.5rem;\n    /* ie11 hack to center vertically */\n    /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n    height: 0;\n  }\n  .mc-button--fit\\@from-l {\n    display: -webkit-inline-box;\n    display: -ms-inline-flexbox;\n    display: inline-flex;\n  }\n  .mc-button--full\\@from-l {\n    display: -webkit-box;\n    display: -ms-flexbox;\n    display: flex;\n  }\n}\n\n@media screen and (min-width: 1280px) {\n  .mc-button--s\\@from-xl {\n    font-size: 0.875rem;\n    line-height: 1;\n    padding: 0.25rem 1rem;\n    min-height: 2rem;\n    min-width: 2rem;\n    /* ie11 hack to center vertically */\n    /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n    height: 0;\n  }\n  .mc-button--m\\@from-xl {\n    font-size: 1rem;\n    line-height: 1;\n    padding: 0.25rem 2rem;\n    min-height: 3rem;\n    min-width: 3rem;\n    /* ie11 hack to center vertically */\n    /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n    height: 0;\n  }\n  .mc-button--l\\@from-xl {\n    font-size: 1.125rem;\n    line-height: 1;\n    padding: 0.25rem 2rem;\n    min-height: 3.5rem;\n    min-width: 3.5rem;\n    /* ie11 hack to center vertically */\n    /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n    height: 0;\n  }\n  .mc-button--fit\\@from-xl {\n    display: -webkit-inline-box;\n    display: -ms-inline-flexbox;\n    display: inline-flex;\n  }\n  .mc-button--full\\@from-xl {\n    display: -webkit-box;\n    display: -ms-flexbox;\n    display: flex;\n  }\n}\n\n@media screen and (min-width: 1920px) {\n  .mc-button--s\\@from-xxl {\n    font-size: 0.875rem;\n    line-height: 1;\n    padding: 0.25rem 1rem;\n    min-height: 2rem;\n    min-width: 2rem;\n    /* ie11 hack to center vertically */\n    /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n    height: 0;\n  }\n  .mc-button--m\\@from-xxl {\n    font-size: 1rem;\n    line-height: 1;\n    padding: 0.25rem 2rem;\n    min-height: 3rem;\n    min-width: 3rem;\n    /* ie11 hack to center vertically */\n    /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n    height: 0;\n  }\n  .mc-button--l\\@from-xxl {\n    font-size: 1.125rem;\n    line-height: 1;\n    padding: 0.25rem 2rem;\n    min-height: 3.5rem;\n    min-width: 3.5rem;\n    /* ie11 hack to center vertically */\n    /* see : https://github.com/philipwalton/flexbugs/issues/231 */\n    height: 0;\n  }\n  .mc-button--fit\\@from-xxl {\n    display: -webkit-inline-box;\n    display: -ms-inline-flexbox;\n    display: inline-flex;\n  }\n  .mc-button--full\\@from-xxl {\n    display: -webkit-box;\n    display: -ms-flexbox;\n    display: flex;\n  }\n}\n.example {\n  text-align: center;\n  margin: 1rem auto;\n}\n',
              },
            },
          },
        ],
      },
    })
  )
})

describe(`Preview Component`, () => {
  test(`check iframe's src`, () => {
    const location = {
      pathname: 'src/docs/Components/Buttons/previews/button-icon-position',
    }
    const preview = mount(
      <Preview location={location} path={'button-icon-position'} />
    )
    const iframe = preview.find('PreviewComponent')
    expect(
      iframe
        .find('iframe')
        .props()
        .src.split('/')
        .pop()
    ).toBe('button-icon-position.html')
  })
})
