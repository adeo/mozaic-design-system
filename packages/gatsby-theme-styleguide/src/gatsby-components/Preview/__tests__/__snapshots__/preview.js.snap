// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview Component check iframe's src 1`] = `
.c19 {
  width: 1.2rem;
  height: 1.2rem;
  box-sizing: border-box;
  cursor: pointer;
  background: rgb(42,39,52);
  box-shadow: none;
  border: none;
  position: absolute;
  right: 1.1rem;
  top: 6px;
  outline: none;
  padding: 0.1rem;
}

.c19 svg {
  overflow: visible;
}

.c18 {
  min-height: 1em;
}

.c18:before {
  display: inline-block;
  width: 3em;
  overflow: hidden;
  font-size: 0.8em;
}

.c17 {
  max-width: 100%;
  overflow: auto;
  margin: 0 !important;
  white-space: pre-wrap;
  border: none !important;
}

.c15 {
  position: relative;
}

.c16 {
  margin-bottom: 0 !important;
  margin-top: 0 !important;
  position: auto !important;
  padding: 1rem 1rem 0.4rem 1rem !important;
}

.c11 {
  background: #f5f5f5;
  padding-left: 10px;
  border: solid 1px #ececec;
  border-top: 0;
}

.c12 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  box-shadow: none;
  border: none;
  line-height: 1.5rem;
  font-weight: bold;
  background: #fff;
  border: solid 1px #ececec;
  border-bottom: 0;
  color: #554f52;
  text-transform: uppercase;
  cursor: pointer;
  outline: none;
  margin: 0;
  border-radius: 4px 4px 0 0;
  margin-top: 6px;
  margin-bottom: -1px;
  padding: 5px 16px;
}

.c13 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  box-shadow: none;
  border: none;
  line-height: 1.5rem;
  font-weight: bold;
  background: transparent;
  border: solid 1px transparent;
  border-bottom: 0;
  color: #554f52;
  text-transform: uppercase;
  cursor: pointer;
  outline: none;
  margin: 0;
  border-radius: 4px 4px 0 0;
  margin-top: 6px;
  margin-bottom: -1px;
  padding: 5px 16px;
}

.c20 {
  background: #f5f5f5;
  color: #554f52;
  padding: 5px 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  border-radius: 0 0 4px 4px;
  border: solid 1px #ececec;
  border-top: 0;
}

.c21 {
  color: #554f52;
  box-shadow: none;
  border: none;
  background: none;
  padding: none;
  margin: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-size: 0.75rem;
  outline: none;
}

.c14 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
  position: relative;
  border-left: solid 1px #ececec;
  max-height: 10vh;
}

.c1 {
  background: #f5f5f5;
  color: #554f52;
  border: solid 1px #ececec;
  border-radius: 4px;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 34px;
  position: relative;
  overflow: visible;
}

.c3 {
  box-sizing: border-box;
  cursor: pointer;
  background: none;
  box-shadow: none;
  border: none;
  outline: none;
}

.c5 {
  box-sizing: border-box;
  cursor: pointer;
  background: none;
  box-shadow: none;
  border: none;
  outline: none;
}

.c2 {
  display: inline-block;
  vertical-align: middle;
  padding: 0.5rem;
  border-right: solid 1px #ececec;
}

.c4 {
  color: #666;
  font-size: 11px;
  margin-right: 10px;
}

.c7 {
  position: relative;
  padding: 1rem 0;
  box-shadow: inset 0 0 0 1px #ececec;
  background-color: #fff;
  width: px;
  min-width: px;
  -webkit-transform: scale( NaN );
  -ms-transform: scale( NaN );
  transform: scale( NaN );
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
}

.c6 {
  background: #f5f5f5;
  border-bottom: solid 1px #ececec;
  position: relative;
  overflow: hidden;
}

.c8 {
  width: 100%;
  margin: 0;
  padding: 0;
  min-height: 90px;
}

.c9 {
  position: absolute;
  top: 5px;
  right: 5px;
}

.c10 {
  position: absolute;
  top: 0;
  left: 0;
  background: #ececec;
  display: inline-block;
  font-size: 11px;
  -webkit-letter-spacing: 0.08em;
  -moz-letter-spacing: 0.08em;
  -ms-letter-spacing: 0.08em;
  letter-spacing: 0.08em;
  color: #666;
  line-height: 1;
  padding: 6px 8px;
}

.c0 {
  position: relative;
}

<div
  className="c0"
>
  <div
    style={
      Object {
        "margin": "2rem 0",
      }
    }
  >
    <div
      className="c1"
    >
      <div>
        <div
          className="c2"
        >
          <button
            className="c3"
            onClick={[Function]}
          >
            <svg
              fill="#554f52"
              height="12"
              style={Object {}}
              viewBox="0 0 16 16"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink"
            >
              <path
                d="M6 6h-6v-6h6v6zm9-6h-6v6h6v-6zm9 0h-6v6h6v-6zm-18 9h-6v6h6v-6zm9 0h-6v6h6v-6zm9 0h-6v6h6v-6zm-18 9h-6v6h6v-6zm9 0h-6v6h6v-6zm9 0h-6v6h6v-6z"
                fill="#554f52"
              />
            </svg>
          </button>
        </div>
        <div
          className="c2"
        >
          <svg
            fill="#554f52"
            height="12"
            style={
              Object {
                "marginRight": "8px",
              }
            }
            viewBox="0 0 16 16"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <path
              d="M0 1v10h16v-10h-16zM15 10h-14v-8h14v8zM10.5 12h-5l-0.5 2-1 1h8l-1-1z"
              fill="#554f52"
            />
          </svg>
          <select
            onChange={[Function]}
          >
            <option
              default={false}
              value="auto"
            >
              auto
            </option>
            <option
              default={false}
            >
              s
            </option>
            <option
              default={false}
            >
              s-medium
            </option>
            <option
              default={false}
            >
              s-large
            </option>
            <option
              default={false}
            >
              s-xlarge
            </option>
            <option
              default={false}
            >
              m
            </option>
            <option
              default={false}
            >
              m-medium
            </option>
            <option
              default={false}
            >
              l
            </option>
            <option
              default={false}
            >
              l-medium
            </option>
            <option
              default={false}
            >
              xl
            </option>
            <option
              default={false}
            >
              xl-medium
            </option>
            <option
              default={false}
            >
              xl-large
            </option>
            <option
              default={false}
            >
              xxl
            </option>
          </select>
        </div>
      </div>
      <div
        className="c2"
      >
        <a
          className="c4"
          href=""
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            fill="#554f52"
            height="12"
            style={Object {}}
            viewBox="0 0 16 16"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <polygon
              id="box"
              points="0,0 6,0 6,2 2,2 2,14 14,14 14,10 16,10 16,16 0,16"
            />
            <polygon
              id="arrow_13_"
              points="8.4,0 16,0 16,7.6 13.1,4.8 8.9,9 7,7.1 11.2,2.9"
            />
          </svg>
        </a>
        <button
          className="c5"
          onClick={[Function]}
        >
          <svg
            fill="#554f52"
            height="12"
            style={Object {}}
            viewBox="0 0 16 16"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <path
              d="M16 0v6.5l-2.5-2.5-3 3-1.5-1.5 3-3-2.5-2.5zM7 10.5l-3 3 2.5 2.5h-6.5v-6.5l2.5 2.5 3-3z"
              fill="#554f52"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="c6"
    >
      <div
        className="c7"
      >
        <iframe
          className="c8"
          frameBorder="0"
          height="90px"
          onLoad={[Function]}
          scrolling="no"
          src=""
        />
      </div>
      <button
        className="c9"
        onClick={[Function]}
      >
        hide options
      </button>
      <span
        className="c10"
      >
        Viewport: 
        px
         
      </span>
    </div>
    <div
      className="c11"
    >
      <button
        className="c12"
        onClick={[Function]}
      >
        html
      </button>
      <button
        className="c13"
        onClick={[Function]}
      >
        css
      </button>
      <button
        className="c13"
        onClick={[Function]}
      >
        scss
      </button>
    </div>
    <div
      className="c14"
    >
      <div
        className="c15"
      >
        <pre
          className="c16 prism-code language-html"
        >
          <code
            className="c17 prism-code language-html"
            style={Object {}}
          >
            <div
              className="c18 token-line"
            >
              <span
                className="token plain"
              >
                not found
              </span>
            </div>
          </code>
        </pre>
        <button
          className="c19"
          onClick={[Function]}
        >
          <svg
            fill="#ececec"
            height="0.8rem"
            style={Object {}}
            viewBox="0 0 16 16"
            width="0.8rem"
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <path
              d="M10 4v-4h-7l-3 3v9h6v4h10v-12h-6zM3 1.414v1.586h-1.586l1.586-1.586zM1 11v-7h3v-3h5v3l-3 3v4h-5zM9 5.414v1.586h-1.586l1.586-1.586zM15 15h-8v-7h3v-3h5v10z"
              fill="#ececec"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="c20"
    >
      <button
        className="c21"
        onClick={[Function]}
      >
        Show more  
        <svg
          fill="#554f52"
          height="12px"
          style={Object {}}
          viewBox="0 0 16 16"
          width="12px"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="M16 8c0-4.418-3.582-8-8-8s-8 3.582-8 8 3.582 8 8 8 8-3.582 8-8zM1.5 8c0-3.59 2.91-6.5 6.5-6.5s6.5 2.91 6.5 6.5-2.91 6.5-6.5 6.5-6.5-2.91-6.5-6.5z"
            fill="#554f52"
          />
          <path
            d="M4.957 5.543l-1.414 1.414 4.457 4.457 4.457-4.457-1.414-1.414-3.043 3.043z"
            fill="#554f52"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
